### ISSUE TITLE
 '_ticker != null' after successfull login

### ISSUE THREAD

**Issue by SalahAdDin** [NearHuscarl/flutter_login#152]
**Describe the bug**
We are using this package to handle our application's login workflow.

We are using _riverpod_ as state manager and we pass our provider's functions as callback.

Althought there is no a visible problem and we can login succesfully, we found the next bug on the terminal at running:
```shell
[+202334 ms] I/flutter (17870): User attributes: {sub: 1f688bb8-726e-4b17-b705-6f93633f4105, email_verified: true, phone_number_verified: true, phone_number: +905373749236, email: luis.alaguna@cloudnesil.com}
[ +430 ms] E/flutter (17870): [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: 'package:flutter/src/animation/animation_controller.dart': Failed assertion: line 488 pos 7: '_ticker != null': AnimationController.reverse() called after AnimationController.dispose()
[   +1 ms] E/flutter (17870): AnimationController methods should not be used after calling dispose.
[   +4 ms] E/flutter (17870): #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:46:39)
[   +1 ms] E/flutter (17870): #1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:36:5)
[   +1 ms] E/flutter (17870): #2      AnimationController.reverse (package:flutter/src/animation/animation_controller.dart:488:7)
[        ] E/flutter (17870): #3      _LoginCardState._submit (package:flutter_login/src/widgets/auth_card.dart:520:29)
[   +6 ms] E/flutter (17870): <asynchronous suspension>
[        ] E/flutter (17870): 
[ +276 ms] E/flutter (17870): [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: setState() called after dispose(): _LoginCardState#dc130(lifecycle state: defunct, not mounted, tickers: tracking 0 tickers)
[        ] E/flutter (17870): This error happens if you call setState() on a State object for a widget that no longer appears in the widget tree (e.g., whose parent widget no longer includes the widget in its build). This error can occur when code calls setState() from a timer or an animation callback.
[        ] E/flutter (17870): The preferred solution is to cancel the timer or stop listening to the animation in the dispose() callback. Another solution is to check the "mounted" property of this object before calling setState() to ensure the object is still in the tree.
[   +1 ms] E/flutter (17870): This error might indicate a memory leak if setState() is being called because another object is retaining a reference to this State object after it has been removed from the tree. To avoid memory leaks, consider breaking the reference to this object during dispose().
[        ] E/flutter (17870): #0      State.setState.<anonymous closure> (package:flutter/src/widgets/framework.dart:1231:9)
[        ] E/flutter (17870): #1      State.setState (package:flutter/src/widgets/framework.dart:1266:6)
[        ] E/flutter (17870): #2      _LoginCardState._submit.<anonymous closure> (package:flutter_login/src/widgets/auth_card.dart:517:7)
[        ] E/flutter (17870): #3      new Future.delayed.<anonymous closure> (dart:async/future.dart:315:39)
[        ] E/flutter (17870): #4      _rootRun (dart:async/zone.dart:1346:47)
[        ] E/flutter (17870): #5      _CustomZone.run (dart:async/zone.dart:1258:19)
[        ] E/flutter (17870): #6      _CustomZone.runGuarded (dart:async/zone.dart:1162:7)
[        ] E/flutter (17870): #7      _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1202:23)
[        ] E/flutter (17870): #8      _rootRun (dart:async/zone.dart:1354:13)
[        ] E/flutter (17870): #9      _CustomZone.run (dart:async/zone.dart:1258:19)
[        ] E/flutter (17870): #10     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:1186:23)
[        ] E/flutter (17870): #11     Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:18:15)
[        ] E/flutter (17870): #12     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:395:19)
[        ] E/flutter (17870): #13     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:426:5)
[        ] E/flutter (17870): #14     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
[        ] E/flutter (17870): 
[+7195 ms] D/InputConnectionAdaptor(17870): The input method toggled cursor monitoring on
[+56382 ms] DevFS: Deleting filesystem on the device (file:///data/user/0/com.cloudnesil.thesis_cancer/code_cache/thesis_cancerVCZBKY/thesis_cancer/)
[  +17 ms] DevFS: Deleted filesystem on the device (file:///data/user/0/com.cloudnesil.thesis_cancer/code_cache/thesis_cancerVCZBKY/thesis_cancer/)
```

**To Reproduce**
1. Run the application.
2. Fulfill credentials.
3. Do login.
4. Go to the terminal and see the error.

**Expected behavior**
It should not show similar bugs on the terminal.

**Screenshots**

**Information (please complete the following information):**
 - Device: Xioami Redmi Note 7
 - Platform: Android 10
 - Flutter version: Flutter 2.0.4 • channel stable • https://github.com/flutter/flutter.git
 - Package version flutter_login: ^2.0.0-nullsafety.0

**Additional context**
Following are the functions we are passing as callback:
```dart
Future<String?> signIn(
      {required String username, required String password}) async {
    try {
      User storedUser = await dataStore.getUserProfileData();
      AmplifyResult result =
          await authRepository.signIn(username: username, password: password);
      if (result.isSuccess)
        await createUserProfile(storedUser);
      else if (result.nextStep == 'CONFIRM_SIGN_IN_WITH_NEW_PASSWORD')
        state = AuthState.requiresConfirmSignIn();
      else
        state = AuthState.error(
            'It was not possible to log in with your credentials.');
    } on LogInUnconfirmedUserFailure {
      User newProfile = User(
          id: uuid.v4(),
          email: username,
          displayName: '',
          role: UserRole.PILOT,
          isConfirmed: false);
      state = AuthState.loggedIn(newProfile);
    } on LogInWithEmailAndPasswordFailure catch (error) {
      state = AuthState.error(error.toString());
    }
  }
```

And the next is who we are passing them as callbacks:
```dart
onLogin: (LoginData data) => context
          .read(authNotifierProvider)
          .signIn(username: data.name, password: data.password),
```

**Comment by sarbogast** [comment]
I'm seeing the same issue on the web

**Comment by SalahAdDin** [comment]
@juliansteenbakker I just see the same error again, this time in a **Xiaomi Redmi Note 7**.

**Comment by noskap** [comment]
Same issue here, iOS and Android

**Comment by juliansteenbakker** [comment]
Fixed in #233
